src/uu/df/src/blocks.rs

```diff
@@ -57,6 +73,37 @@ fn to_magnitude_and_suffix_1024(n: u128) -> Result<String, ()> {
     Err(())
 }
 
+/// Convert a number, except multiples of 1024, into a string like "12kB" or "34MB".
+///
+/// Powers of 1000 become "1kB", "1MB", "1GB", etc.
+///
+/// The returned string has a maximum length of 5 chars, for example: "1.1kB", "999kB", "1MB".
+fn to_magnitude_and_suffix_not_powers_of_1024(n: u128) -> Result<String, ()> {
```

Comment:
Seems like there is never an `Err` returned, so does this need to return a `Result`? Maybe there wants to be an `Err` if `n` is larger than 10^27, I'm not sure.

Reply:
Yes, you are right, technically no `Result` is needed. I used it for consistency reasons because `to_magnitude_and_suffix()` returns a `Result` and the following looks cleaner
```rust
fn to_magnitude_and_suffix(n: u128) -> Result<String, ()> {
   if n % 1024 == 0 {    
       to_magnitude_and_suffix_1024(n)
   } else {
       to_magnitude_and_suffix_not_powers_of_1024(n)
   }
}
```
than
```rust
fn to_magnitude_and_suffix(n: u128) -> Result<String, ()> {
   if n % 1024 == 0 {    
       to_magnitude_and_suffix_1024(n)
   } else {
       Ok(to_magnitude_and_suffix_not_powers_of_1024(n))
   }
}
```
I don't think an `Err` is necessary because there is already a "block size too large" error when parsing the provided argument.